EKF/control.cpp

```diff
@@ -730,9 +730,11 @@ void Ekf::controlHeightFusion()
 				}
 			}
 
-		} else if (_baro_data_ready && !_baro_hgt_faulty && !_in_range_aid_mode) {
+		} else if (_baro_data_ready && !_baro_hgt_faulty &&
```

Comment:
Range data is not available every iteration (_range_data_ready = false). Because of that it would switch back and forth between range finder and baro.